### PR TITLE
CLN: Remove future errors and warnings

### PR DESCRIPTION
**Pull request by bashtage**
Convert all ptp uses to np.ptp due to deprecation on DataFrames and Series
Squeeze multi-dimensional arrays to avoid stricter rules for array creationin NumPy 1.6
Replace read_table with read_csv due to deprecation in pandas

### CODE REVIEW

File: statsmodels/stats/tests/test_anova.py [hunk 1]
```diff
@@ -72,7 +72,7 @@
 """)
 
 kidney_table.seek(0)
-kidney_table = read_table(kidney_table, sep="\s+").astype(int)
+kidney_table = read_csv(kidney_table, sep="\s+", engine='python').astype(int)
```
Comment (jbrockmendel) [inline statsmodels/stats/tests/test_anova.py]: Does the engine make a difference here?

**Comment by jbrockmendel** [comment]
LGTM.

**Comment by jbrockmendel** [comment]
the Coveralls measurement looks wrong, I don’t see how it could go down 0.03% with nothing relevant changed.  Is there any way to get rid of the red x?  It makes it harder to see which PRs need which kind of attention.

File: statsmodels/stats/tests/test_anova.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/stats/tests/test_anova.py]: engine prevents a warning since regex isn't available in c

**Comment by bashtage** [comment]
Coverage decreased since I deleted a lot of lines that are no longer needed. Codecov is better since it report overall but also coverage of changed lines, so that you can get 100% on the latter even if the former drops.

**Comment by jbrockmendel** [comment]
> Coverage decreased since I deleted a lot of lines that are no longer needed

Did you?  In the diff it looks like a net change of zero in each affected file.

**Comment by bashtage** [comment]
> Did you? In the diff it looks like a net change of zero in each affected file.

Yes, it does seem random.  Another reason to switch to codecov.

**Comment by ChadFulton** [comment]
This LGTM, but I'm not up on all the details, so will push plan to merge until this weekend, unless there are concerns.

**Comment by bashtage** [comment]
There are errors that will start appearing in an allowed failure test run against NumPY 1.16, SciPy 1.2 and pandas 0.24 nightlys.

**Comment by josef-pkt** [comment]
(appveyor has mostly caught up with test runs)

LGTM
we can still use x.ptp() method when we know and have enforced that it's a numpy array
(I like the method versions of array functions)

merging
@bashtage Thank you

**Comment by josef-pkt** [comment]
BTW: compatibility fixes are `MAINT` and not `CLN` commits

**Pull request merged**